kong/plugins/proxy-cache/cache_key.lua

```diff
@@ -108,8 +109,8 @@ function _M.build_cache_key(consumer_id, route_id, method, uri,
   local params_digest  = params_key(params_table, conf)
   local headers_digest = headers_key(headers_table, conf)
 
-  return md5(fmt("%s|%s|%s|%s|%s", prefix_digest, method, uri, params_digest,
-                                   headers_digest))
+  return sha256_hex(fmt("%s|%s|%s|%s|%s", prefix_digest, method, uri,
+                          params_digest, headers_digest))
```

Comment:
i did a quick perf test since sha256 is known slower than md5; for 100000 iterrations:
sha256 took: 0.042999982833862
md5 took: 0.016000032424927

it doesn't seem too terrible though. on the side, i'm thinking if we can make this function same as the `fips_hash` function that only choose sha256 for fips mode? @samugi  how do you think?

Reply:
I should have added some background on this choice, thanks for the reminder @fffonion.

@bungle and I talked about this yesterday and based on our tests it seems md5 would not provide any performance benefits over sha256, at least on our machines. The test results were actually showing better numbers for sha256 for both of us, with a ~30% improvement over md5 for me and a ~50% improvement for @bungle. 

We used `resty.openssl.digest` with `md5` and `sha256` algorithms, on 10.000.000 iterations, the results are below:

```
@bungle:
md5: 1400ms
sha256: 658ms

mine:
md5: 961ms
sha256: 694ms
```

Reply:
Discussed with @samugi offline, i agree performance is no longer a concern because of the CPU accelaration for sha256. Though from my test ngx.md5_bin is still faster than openssl.md5 and openssl.sha256 since Nginx uses its own md5 implementaion, but that may indicate something can be improved in lua-resty-openssl.
Consider this resolved.